src/Akka.Streams.Kafka/Stages/CommittableConsumerStage.cs

```diff
@@ -192,23 +137,65 @@ private void MessagesReceived(ConsumerRecord<K, V> message)
             }
         }
 
+        private void HandleOnPartitionsAssigned(IConsumer<K, V> consumer, List<TopicPartition> list)
+        {
+            _partitionsAssigned(list);
+        }
+        
+        
         private void PartitionsAssigned(IEnumerable<TopicPartition> partitions)
         {
             Log.Debug($"Partitions were assigned: {_consumer.Name}");
             _consumer.Assign(partitions);
             _assignedPartitions = partitions;
         }
 
-        private void PartitionsRevoked(IEnumerable<TopicPartition> partitions)
+        private void HandleOnPartitionsRevoked(IConsumer<K, V> consumer, List<TopicPartitionOffset> list)
+        {
+            _partitionsRevoked(list);
+        }
+
+        private void PartitionsRevoked(IEnumerable<TopicPartitionOffset> partitions)
         {
             Log.Debug($"Partitions were revoked: {_consumer.Name}");
             _consumer.Unassign();
             _assignedPartitions = null;
         }
+        
+        private void HandleConsumeError(object sender, Error error)
+        {
+            Log.Error(error.Reason);
+            var exception = new KafkaException(error);
+            switch (_decider(exception))
+            {
+                case Directive.Stop:
+                    // Throw
+                    _completion.TrySetException(exception);
+                    FailStage(exception);
+                    break;
+                case Directive.Resume:
+                    // keep going
+                    break;
+                case Directive.Restart:
+                    // keep going
+                    break;
+            }
+        }
 
         private void PullQueue()
         {
-            _consumer.Poll(_settings.PollTimeout);
+            try
+            {
+                var message = _consumer.Consume(_settings.PollTimeout);
+                if (message == null)
+                    return;
+                
+                HandleMessage(message);
+            }
+            catch (ConsumeException ex)
+            {
+                HandleError(ex.Error);
+            }
 
             if (!_isPaused && _buffer.Count > _settings.BufferSize)
```

Comment:
Guess we can get rid of the `_buffer` here?

Reply:
Sure, I will